substrate/frame/election-provider-multi-block/src/lib.rs

```diff
@@ -2356,6 +2444,53 @@ mod election_provider {
 		});
 	}
 
+	#[test]
+	fn elect_advances_phase_even_on_error() {
+		// Use Continue fallback to avoid emergency phase when both primary and fallback fail.
+		// If we don't do this, when we simulate below the page lookup to fail, we would go to
+		// `Phase::Emergency` and not to `Phase::Export(_)`.
+		ExtBuilder::full().fallback_mode(FallbackModes::Continue).build_and_execute(|| {
+			roll_to_signed_open();
+
+			// load a solution into the verifier
+			let paged = OffchainWorkerMiner::<Runtime>::mine_solution(Pages::get(), false).unwrap();
+
+			load_signed_for_verification_and_start_and_roll_to_verified(99, paged, 0);
+
+			// move to Done phase
+			roll_to_done();
+
+			// First elect call should succeed
+			let result1 = MultiBlock::elect(2);
+			assert!(result1.is_ok());
+			assert_eq!(MultiBlock::current_phase(), Phase::Export(1));
+
+			// Now we need to simulate a scenario where get_queued_solution_page(1) returns None
+			// even though we have a queued solution. This should never happen under normal
+			// circumstances so it's considered an exceptional case (e.g. partial storage corruption
+			// or simply a bug).
+			// One way to simulate this is to manipulate the round number to create a mismatch
+			// between what's expected and what's stored.
+			// Let's advance the round manually to create a mismatch
+			use crate::Round;
```

Comment:
So let's zoom out. When can `T::Verifier::get_queued_solution_page` fail? 

There are two scenarios here.

1. Full submissions aka Signed submissions: As per the other PR, they are full, and even empty pages are cased to `Default::default`. 
2. Partial submissions, aka Unsigned submissions.

To test this, please use `OffchainWorkerMiner::<Runtime>::mine_solution(Pages::get() - 1, false).unwrap();` rather than artificially bumping the round. This is too hacky and we can simulate a realistic scenario for it.